aries_cloudagent/transport/pack_format.py

```diff
@@ -218,3 +300,106 @@ def get_recipient_keys(self, message_body: Union[str, bytes]) -> List[str]:
             )
 
         return recipient_keys
+
+
+class V2PackWireFormat(BaseWireFormat):
+    """DIDComm V2 message parser and serializer."""
+
+    async def parse_message(
+        self,
+        session: ProfileSession,
+        message_body: Union[str, bytes],
+    ) -> Tuple[dict, MessageReceipt]:
+        """Parse message."""
+
+        messaging = session.inject(DIDCommMessaging)
+
+        LOGGER.debug("HIT V2 PACK FORMAT .parse_message()")
+        LOGGER.debug(message_body)
+
+        receipt = MessageReceipt()
+        receipt.in_time = time_now()
+        receipt.raw_message = message_body
+        receipt.didcomm_version = DIDCommVersion.v2
+
+        message_dict = None
+        message_json = message_body
+
+        if not message_json:
+            raise WireFormatParseError("Message body is empty")
+
+        try:
+            message_dict = json.loads(message_json)
+        except ValueError:
+            raise WireFormatParseError("Message JSON parsing failed")
+        if not isinstance(message_dict, dict):
+            raise WireFormatParseError("Message JSON result is not an object")
+
+        wallet = session.inject_or(BaseWallet)
+        if not wallet:
+            LOGGER.error("No wallet found")
+            raise WalletNotFoundError()
+
+        # packed messages are detected by the absence of type
+        if "type" not in message_dict:
+            try:
+                message_unpack = await messaging.unpack(message_json)
+            except CryptoServiceError:
+                LOGGER.debug("Message unpack failed, falling back to JSON")
+            else:
+                # Set message_dict to be the dictionary that we unpacked
+                message_dict = message_unpack.message
+
+        if message_unpack.sender_kid:
+            receipt.sender_verkey = message_unpack.sender_kid.split("#")[0]
+            receipt.recipient_verkey = message_unpack.recipient_kid.split("#")[0]
+            pass
```

Comment:
The `pass` is not needed

Reply:
Correct. It was added as I was typing out the code and testing to prevent the code from failing. It can be removed 

One of the downsides of Python is the need for pass in empty blocks xP